### PR TITLE
[slack] Add `One Click Install` API and connect with frontend

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Created backend API for the `One Click Install` which creates Slack App Install link with the given hostname

## How was this patch tested?

- Currently via adding unit tests and successfully running it

### CODE REVIEW

File: desktop/core/src/desktop/lib/botserver/api.py [hunk 1]
```diff
@@ -57,6 +61,27 @@ def send_message(request):
...
+
+    data['features']['unfurl_domains'] = [hostname]
+    data['settings']['event_subscriptions']['request_url'] = 'https://' + hostname + '/desktop/slack/events/'
+
+    changed_data = yaml.safe_dump(data)
+    install_link += quote_plus(changed_data)
+
+    response['link'] = install_link
+
+  return JsonResponse(response)
```
Comment (romainr) [inline desktop/core/src/desktop/lib/botserver/api.py:83]: Would it be simpler to just redirect to the url?

**Review by romainr — APPROVED** [review]
Nice!

File: desktop/core/src/desktop/lib/botserver/api.py (hunk 1 above)
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/api.py:83]: @romainr  trying with `return redirect(install_link)` but when testing `response.status_code` , its failing with 403 status code. Not sure where am I missing for a correct redirect to the install link. Any hints?

**Comment by Harshg999** [comment]
Connected with the frontend. Would like your suggestions and feedback! @romainr @JohanAhlen 
- Need to have a catchy image and a small paragraph still though

Here's a quick preview! 

one-click

File: docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html [hunk 2]
```diff
@@ -174,4 +156,39 @@ <h2>Find and connect your data</h2>
...
   </section>
+  <section class="switchable feature-large switchable--switch bg--secondary">
+    <div class="container">
+      <div class="row justify-content-around">
+        <div class="col-md-6">
+          <img alt="Image" class="border--round box-shadow-wide" src="img/databases.svg" />
+        </div>
+        <div class="col-md-6 col-lg-5">
+          <div class="switchable__text input-append">
+            <h2>Get your own SQL assistant</h2>
```
Comment (romainr) [inline docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html:167]: Need to be more explicit that this is about Slack and unfurling and link to install etc...

File: docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html [hunk 3]
```diff
@@ -174,4 +156,39 @@ <h2>Find and connect your data</h2>
...
+            <p class="lead">
+              Assist users with their SQL queries and have better collaboration with your team.
+            </p>
+            <form autocomplete="off" onsubmit="return check(event)">
```
Comment (romainr) [inline docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html:171]: Not sure why not just doing a POST on the form? (no js needed this way)

**Review by romainr — COMMENTED** [review]
Nice!

Catchy image --> simple Slack logo?

**Comment by Harshg999** [comment]
Screenshot 2021-05-30 at 2 17 34 PM

File: docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html [hunk 4]
```diff
@@ -174,4 +156,45 @@ <h2>Find and connect your data</h2>
...
   </section>
+  <section class="switchable feature-large switchable--switch bg--secondary">
+    <div class="container">
+      <div class="row justify-content-around">
+        <div class="col-md-6">
+          <img alt="Image" class="border--round box-shadow-wide" src="img/hue_slack_integration.png" />
+        </div>
+        <div class="col-md-6 col-lg-5">
+          <div class="switchable__text">
+            <h2>Get your own SQL assistant</h2>
```
Comment (romainr) [inline docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html:167]: "Install the Slack App!"

More explicit probably better for now?

File: docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html [hunk 5]
```diff
@@ -174,4 +156,45 @@ <h2>Find and connect your data</h2>
...
+  function check(event) {
+    event.preventDefault();
+
+    const url = 'https://demo.gethue.com/desktop/slack/api/install/?hostname=' + document.getElementById('hostname').value;
+    fetch(url)
+      .then(response => response.json())
+      .then(response => {
+        window.location.replace(response['link']);
+      });
+  }
```
Comment (romainr) [inline docs/gethue/themes/stack-hue-theme/layouts/partials/home.en.html:199]: PS: would no need any of this if using a regular html POST form..

**Review by romainr — APPROVED** [review]

**Pull request merged**